presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -878,6 +878,12 @@ public void testInPredicateWithCasts()
                         cast(C_SMALLINT, BIGINT),
                         new InListExpression(ImmutableList.of(toExpression(1L, BIGINT)))),
                 withColumnDomains(ImmutableMap.of(C_SMALLINT, Domain.singleValue(SMALLINT, 1L))));
+
+        ExtractionResult result = fromPredicate(new InPredicate(
```

Comment:
strange. I would expect narrowing casts to work for integer types. Could you create an issue for that?